packages/ui/src/components/Typography/Typography.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { ReactNode, ReactText } from 'react';
+import { Box, forwardRef, HTMLChakraProps } from '@chakra-ui/react';
+
+type TypographyVariant = 'span' | 'div' | 'p' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+
+type TypographyFontSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl' | '5xl' | '6xl' | '7xl';
+
+type TypographyFontWeight = 'regular' | 'medium';
+
+type RemoveCommonValues<Type, TOmit> = {
+  [Property in keyof Type]: TOmit extends Record<Property, infer U> ? Exclude<Type[Property], U> : Type[Property];
+};
+
+type Id<Type> = Record<string, unknown> & { [P in keyof Type]: Type[P] };
+type ConditionalProps<Type, TKey extends keyof TCase, TCase extends Partial<Type>> =
+  | Id<Omit<Type, keyof TCase> & TCase>
+  | Id<RemoveCommonValues<Type, Pick<TCase, TKey>>>;
+
+export interface TypographyProps extends HTMLChakraProps<'div'> {
+  children: ReactText | ReactText[] | ReactNode | ReactNode[];
+  as?: TypographyVariant | 'a';
+  size?: TypographyFontSize;
+  weight?: TypographyFontWeight;
+}
+export interface TypographyPropsForAnchor extends TypographyProps {
+  href: string;
+  as: 'a';
+}
+
+export type IsAnchor = ConditionalProps<TypographyProps, 'as', TypographyPropsForAnchor>;
+
+const anchorStyle = {
+  color: 'brand.500',
+  textDecoration: 'underline',
+  _hover: { color: 'brand.600' },
+  _active: { color: 'brand.700' },
+};
+
+export const Typography: React.FC<IsAnchor> = forwardRef(
+  ({ as = 'div', size = 'md', weight = 'regular', children, ...props }, ref) => {
+    const stylesForLink = as === 'a' && anchorStyle;
+    return (
+      <Box as={as} fontSize={size} ref={ref} fontWeight={weight} {...props} {...stylesForLink}>
```

Comment:
Wait, `stylesForLink` could be an object or `false` so in many cases you're destructuring the `false` value. I see it works without any error, but I'm thinking why? 😅

Reply:
The magic of JavaScript;-) I can change it to a ternary operator.

Reply:
We can discuss it on Mondays daily. For me, your current solution works fine and it's shorter so you can leave it 😄

Reply:
As a curiosity, there is the reason, why it works. It's a result of how Babel transpiles spread operators
<img width="874" alt="CleanShot 2021-07-03 at 22 05 53@2x" src="https://user-images.githubusercontent.com/22938071/124365845-09ccb280-dc4b-11eb-8ff0-f83a1803553a.png">